### PR TITLE
Jheaser tree page

### PR DESCRIPTION
**Pull request by panzey**
The test data will dynamically generate on refresh of the page, to show how the data may look in a multitude of variations!

### CODE REVIEW

File: extension/js/tree.js [hunk 1]
```diff
@@ -181,14 +205,13 @@ function populateDataIntoCiteBox(citeBox, data) {
...
-	// this function currently only draws lines on the observable page.
-	// changing screen resolution will misplace lines, and scrolling down will reveal an absence of lines.
-	// this should be modified to fix these two issues.
+	// changing screen resolution will misplace lines.
 	var currentCiteBox, parentCiteBox, currentCite, parentCite, currentx, currenty, parentx, parenty, canvasContext;
-	// set canvasElement bounds to observable window
-	deepCite.canvasElement.width = window.innerWidth;
-	deepCite.canvasElement.height = window.innerHeight;
+	// set canvasElement bounds to scrollable screen size.
+	deepCite.canvasElement.width = $(document).width();
```
Comment (panzey) [inline extension/js/tree.js:218]: a use for jquery... i didn't want to admit it

File: extension/js/tree.js [hunk 2]
```diff
@@ -166,6 +159,37 @@ function groupCiteData(data) {
...
+	data.forEach(function (rowSubset) {
+		// each rowSubset is a row of cites
+		subset = [];
+		if (rowSubset.length > 1) {
+			// if there's multiple cites on this row, we need to sort by parentCiteID to keep lines from crossing
+			// problem is, the above row may not be citeID ascending, so we can't order by sort(a - b)
+			previousSortedRowIDs = [];
+			// grab the parent row's citeIDs.  it's sorted correctly at this point.
+			// we can safely assume there will be an array of at least one item here.
+			sortedData[sortedData.length - 1].forEach(function (cite) {
```
Comment (connorjoleary) [inline extension/js/tree.js:173]: Do the cite IDs need to be in order? Right now I am just grabbing a random uuid

**Review by connorjoleary — APPROVED** [review]

File: extension/js/tree.js (hunk 2 above)
Comment (panzey) [reply extension/js/tree.js:173]: i believe that will be totally fine!  we aren't sorting on any int value, so the id could theoretically be any data type as long as it's unique

Comment (connorjoleary) [reply extension/js/tree.js:173]: Could we sort on score instead?

Comment (panzey) [reply extension/js/tree.js:173]: This sort function is sorting row subsets by parentCiteID, but we aren't sorting the cites any further than that -- I think it should be easy to implement around line 178.  Let me do that quick

File: extension/js/tree.js [hunk 3]
```diff
@@ -175,9 +175,16 @@ function sortCiteData(data) {
...
 			previousSortedRowIDs.forEach(function (parentCiteID) {
-				subset.push(...rowSubset.filter(function (cite) {
+				// find the cites that share a parentCiteID
+				unsortedSubset = rowSubset.filter(function (cite) {
 					return cite.parentCiteID === parentCiteID;
-				}));
+				});
+				// order these cites by score, highest to lowest going from left to right
+				sortedSubset = unsortedSubset.sort(function (a, b) {
+					return b.score - a.score;
```
Comment (panzey) [inline extension/js/tree.js:184]: unlike citeID, we are now reliant on score being an int value.  But, they now sort on score!

**Pull request merged**